tools/eslint-plugin/dist/rules/public-extension-exports.js

```diff
@@ -0,0 +1,236 @@
+  [ts.SyntaxKind.ExportDeclaration]: "export",
+  [ts.SyntaxKind.NamespaceExportDeclaration]: "namespace export",
+  [ts.SyntaxKind.VariableStatement]: "variable statement"
+}
+
+var firstRun = true;
+
+/**
+ * This rule prevents the exporting of extension APIs that not not meet certain release tags.
+ */
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Prevent the exporting of extension APIs that not meet certain release tags.",
+      category: "TypeScript",
+    },
+    messages: {
+      forbidden: `{{kind}} "{{name}}" without one of the release tags "{{releaseTags}}".`,
+    },
+    schema: [
+      {
+        type: "object",
+        additionalProperties: false,
+        properties: {
+          releaseTags: {
+            type: "array",
+            uniqueItems: true,
+            items: {
+              type: "string",
+              enum: ["public", "beta", "alpha", "internal"]
+            }
+          },
+          outputApiFile: {
+            type: "boolean"
+          }
+        }
+      }
+    ]
+  },
+
+  create(context) {
+    const parserServices = getParserServices(context);
+
+    const releaseTags = (context.options.length > 0 && context.options[0].releaseTags) || ["public"];
+    const extensionApiTag = "extensionApi"; // SWB temporary extension tag name
+
+    const outputApiFile = (context.options.length > 0 && context.options[0].outputApiFile) || false;
+    const apiFilePath = "./lib/GeneratedExtensionApi.csv";
+
+    if (firstRun) {
+      firstRun = false;
+      if (outputApiFile) {
+        // create/clear output api file on first run
+        fs.writeFileSync(apiFilePath, "");
+      }
+    }
+
+    function addToApiList(declaration, jsDocExtensionTag) {
+      if (!outputApiFile) {
+        return;
+      }
+
+      // Separate interfaces, items marked as real, enums, and others (treated as types)
+      let trailer = ",type\n";
+      if (declaration.kind === ts.SyntaxKind.EnumDeclaration) {
+        trailer = ",enum\n";
+      }
+      else if (declaration.kind === ts.SyntaxKind.InterfaceDeclaration) {
+        trailer = ",interface\n";
+      }
+      else if (jsDocExtensionTag.comment && jsDocExtensionTag.comment.toLowerCase().includes("real")) {
+        trailer = ",real\n";
+      }
+
+      if (declaration.kind === ts.SyntaxKind.VariableStatement) {
+        fs.writeFileSync(apiFilePath, declaration.declarationList.declarations[0].symbol.escapedName + trailer, { flag: "a" });
+        return;
+      }
+
+      fs.writeFileSync(apiFilePath, declaration.symbol.escapedName + trailer, { flag: "a" });
+    }
+
+    function getFileName(parent) {
+      let currentParent = parent;
+      while (currentParent) {
+        if (currentParent.fileName !== undefined)
+          return currentParent.fileName;
+        currentParent = currentParent.parent;
+      }
+      return undefined;
+    }
+
+    function isLocalFile(declaration) {
```

Comment:
What is this for?

Reply:
I believe this prevents the linter from attempting to lint anything in the node modules folder